mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,786 @@
+    local_sparkml_model_path = str(Path(local_mlflow_model_path) / flavor_conf["model_data"])
+    return _load_model(
+        model_uri=sparkml_model_uri,
+        dfs_tmpdir_base=dfs_tmpdir,
+        local_model_path=local_sparkml_model_path,
+    )
+
+
+def _load_pyfunc(path, spark=None):
+    """
+    Load PyFunc implementation. Called by ``pyfunc.load_model``.
+    :param path: Local filesystem path to the MLflow Model with the ``johnsnowlabs`` flavor.
+    :param spark: Optionally pass spark context when using pyfunc as UDF. required, because we cannot fetch the Sparkcontext inside of the Workernode which executes the UDF
+    :return:
+    """
+    return _PyFuncModelWrapper(_load_model(model_uri=path), spark if spark else _get_or_create_sparksession(path))
+
+
+def _get_or_create_sparksession(model_path=None):
+    """
+    1. Check if SparkSession running and get it
+    2. If none exists, create a new one using jars in model_path
+    3. If model_path not defined, rely on nlp.start() to create a new
+    one using johnsnowlabs Jar resolution method
+    See https://nlp.johnsnowlabs.com/docs/en/jsl/start-a-sparksession
+    and https://nlp.johnsnowlabs.com/docs/en/jsl/install_advanced
+    :param model_path:
+    :return:
+    """
+    _validate_env_vars()
+    from mlflow.utils._spark_utils import _get_active_spark_session
+    spark = _get_active_spark_session()
+    if spark is None:
+        spark_conf = {}
+        spark_conf["spark.python.worker.reuse"] = "true"
+        os.environ['PYSPARK_PYTHON'] = sys.executable
+        os.environ['PYSPARK_DRIVER_PYTHON'] = sys.executable
+        if model_path:
+            jar_paths, license_path = _fetch_deps_from_path(model_path)
+            # jar_paths += get_mleap_jars().split(',')  # TODO when to load MLleap Jars
+            if license_path:
+                with open(license_path) as f:
+                    loaded_license = json.load(f)
+                    os.environ.update({k: str(v) for k, v in loaded_license.items() if v is not None})
+                    os.environ['JSL_NLP_LICENSE'] = loaded_license['HC_LICENSE']
+            _logger.info("Starting a new Session with Jars: ", jar_paths)
+            spark = nlp.start(nlp=False, spark_nlp=False, jar_paths=jar_paths, json_license_path=license_path,
+                              create_jsl_home_if_missing=False, spark_conf=spark_conf)
+        else:
+            spark = nlp.start()
+    return spark
+
+
+def _fetch_deps_from_path(local_model_path):
+    if _JOHNSNOWLABS_MODEL_PATH_SUB not in local_model_path:
+        local_model_path = Path(local_model_path) / _JOHNSNOWLABS_MODEL_PATH_SUB / 'jars.jsl'
+    else:
+        local_model_path = Path(local_model_path) / 'jars.jsl'
+
+    jar_paths = [str(local_model_path / file) for file in local_model_path.iterdir() if file.suffix == '.jar']
+    license_path = [str(local_model_path / file) for file in local_model_path.iterdir() if file.suffix == '.json']
```

Comment:
`file.suffix == '.json'` can match any json files.

Reply:
resolved in https://github.com/mlflow/mlflow/pull/8556/commits/76d55dcdd647b6d183e933e3f7802995c95fc2fc